cmd/server/main.go

```diff
@@ -2,26 +2,84 @@ package main
 
 import (
 	"encoding/json"
+	"flag"
 	"fmt"
 	"io"
 	"log"
+	"net"
 	"net/http"
+	"os"
 	"time"
 
+	"github.com/joho/godotenv"
+
 	"github.com/benchttp/engine/internal/configparse"
 	"github.com/benchttp/engine/runner"
 )
 
 const (
-	port = "8080"
+	ReadySignal = "READY"
+)
+
+var (
+	stdout = log.New(os.Stdout, "", 0)
+	stderr = log.New(os.Stderr, "", log.LstdFlags)
 )
 
 func main() {
-	addr := ":" + port
-	fmt.Println("http://localhost" + addr)
+	useAnyPort := flag.Bool("any-port", true, "use any available port allocated by the os")
+	flag.Parse()
+
+	var p string
+	if !*useAnyPort {
+		err := godotenv.Load("./.env.development")
+		if err != nil {
+			log.Println(err)
+		}
+		p = os.Getenv("VITE_ENGINE_PORT")
+	} else {
+		p = "0"
+	}
+
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:"+p)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Why not use `stderr` defined above? (same L51)
```suggestion
		stderr.Fatal(err)
```